### ISSUE TITLE
Komodo 11 no longer able to select lines by clicking in the left margin 

### ISSUE THREAD

**Issue by jimtut** [Komodo/KomodoEdit#3489]
### Short Summary
In Komodo 10 and earlier, when you click in the lift margin on top of the line number, that entire line is "selected".  If you click and hold, you can drag up or down to select adjacent lines.

In Komodo 11, this feature was removed, and Komodo 11 now creates a breakpoint, something that Komodo 10 and earlier did by clicking to the left of the line number, leaving room for the "select line" functionality when clicking directly on the line number.

Please consider adding this feature back.  It's extremely useful when copying/pasting, reformatting/indenting code, etc.

### Steps to Reproduce
Open any file in Komodo.  Click on the line number in the left-hand margin.

### Expected results
The entire row should be selected.

### Actual results
A breakpoint is created.

### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?* 11.0.2
*Operating System (and version)?* Mac OS X and Windows (I tested it on both)

**Comment by Naatan** [comment]
This is by design, the usage of the margin is explained when you hover over it. Unfortunately we cannot facilitate all possible options for margin clicks, we have to make choices.

**Comment by jimtut** [comment]
Understood.  It's too bad for me, though, as it's something I used the margin for quite a lot.  I'll have to stick to Komodo 10 for now...

**Comment by weiqk** [comment]
The person who introduced this bug should be fired. This is a serious bug that makes komodo no longer great. Using komodo now is as uncomfortable as eating shit.

**Comment by jimtut** [comment]
Yeah, it’s truly unfortunate. I just reimaged my PC and thought I’d give Komodo 11 another try. My god, it’s really bad!  Besides this change, they have that whole weird margin symbol for every line that has a print statement!  I really do not toggle my prints on and off, so this is useless and a LOT of clutter. 

I went back to version 10 again. I’ve bought every release since 6 I think, but I won’t buy any more unless they give the ability to customize how the margin behaves so I can make it like every other editor I use...

**Comment by th3coop** [comment]
This is fixed in 12:
https://github.com/Komodo/KomodoEdit/issues/3786
which you can access here:
https://community.komodoide.com/t/komodo-12-preview-now-available/4830

@jimtut if you have more issues, please report them.  They won't get addressed without reports.  I can't imagine you went back to 10 just because of debug symbols in the margin.  If you have reported them, please reference them for us (and others reading this).  Thanks for the feedback.

**Comment by dbotham** [comment]
Not to raise a dead horse, but, I am running Komodo IDE, version 12.0.1, build 91869, and the left margin allows me to select only a single line.  Cannot select a block of lines like before... (I would love for someone to slap me on the forehead on this one) Not useful. I grab chunks of lines all the time.  A complete rational and write should be required before allow someone to code out of existence something like WISIWIG editing features we fought so hard to get back in Word Perfect 5.2 :)

**Comment by th3coop** [comment]
@dbotham What's preventing you from selecting the lines in the editor?  I don't understand the issue.

**Comment by jimtut** [comment]
@dbotham, I was just thinking of leaving the same feedback!

I gave up on Komodo 11 because you couldn't select a line of text by clicking in the left margin.  I stuck with Komodo 10 for years because of this.  I was SO glad to see it come back in Komodo 12, but you're right, they missed a key feature.

Without trying to record a movie of my screen, let me explain what EVERY single other editor I've used in the last 20+ years does.  This includes MS Word, and I know that's not a code editor, but it's pretty ubiquitous.

1. If you double-click on text, you highlight the whole word.  If you double-click AND hold the mouse down on the 2nd click, and drag the mouse, you will continue to highlight all of the whole words that you mouse over.

1. If you triple-click on text, you highlight the whole line (paragraph in MS Word).  If you triple-click AND hold the mouse down on the 3rd click, you will continue to highlight all of the LINES that you mouse over (paragraphs in MS Word).

1. If you click in the left margin, you highlight the whole line (even in MS Word).  If you click in the left margin AND hold the mouse down on the click, and drag the mouse, you will continue to highlight all of the lines that you mouse over.

Every editor does these things.  Komodo added part of the last item back in v12, but it is still NOT doing the last part of that - if you click in the left margin, hold-down and drag, nothing happens except the original line being highlighted.

It would get GREAT if this could get added back into Komodo 12 (this worked in v10), as it's a pretty natural thing to do when copying or deleting large blocks of code.  I'd be willing to submit a new feature request, if that would help!

**Comment by dbotham** [comment]
> @dbotham What's preventing you from selecting the lines in the editor? I don't understand the issue.

The source code, I would imagine... 

- Click in left margin => line is selected
- Hold mouse button down and drag down => original (single) line stays seleted

I.e. cannot select lines (notice the 's') from the left margin... Just a single line... Not useful.

I can understand choices need to be made about how the left margin can be used, but, dropping drag select?  That has to be a mistake. The "Fix" for /Select Lines/ what /Select Line/... Hmmmmm...

**Comment by th3coop** [comment]
Oh yep, you guys are correct #3786 shouldn't close this.

**Comment by jeffbeining** [comment]
2021 now, had my hard drive die so I was forced to install Komodo 12. Yeah, not having multiline select from the margin is stupid. Please add this back. There's no other functionality that happens on the line numbers, I don't see why you couldn't re-add it.
